exporters/FusionRobotExporter/Source/AddIn/Analytics.cpp

```diff
@@ -0,0 +1,126 @@
+#include "Analytics.h"
+#include <nlohmann/json.hpp>
+
+using namespace std;
+using namespace web;
+using namespace web::http;
+using namespace web::http::client;
+
+using namespace adsk::core;
+using namespace SynthesisAddIn;
+
+utility::string_t toUrlString(const utility::string_t in)
+{
+	return web::uri::encode_uri(in);
+}
+
+utility::string_t removeSpaces(utility::string_t in)
+{
+	in.erase(remove_if(in.begin(), in.end(), isspace), in.end());
+	return in;
+}
+
+std::string Analytics::clientId;
+
+void Analytics::setClientID()
+{
+	std::string jsonId = "";
+	try {
+		std::ifstream t("SynthesisAddInSettings.json"); // TODO: settings manager class
+		std::string jsonStr((std::istreambuf_iterator<char>(t)),
+			std::istreambuf_iterator<char>());
```

Comment:
I didn't see a way to save document-independent settings in the fusion api, hence the `SynthesisAddInSettings.json` file. If more plugin-specific settings are needed in the future, a dedicated class should be made to manage the file.